test/schemaRefresh.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const { test, t } = require('tap')
+
+const FakeTimers = require('@sinonjs/fake-timers')
+const { setImmediate } = require('node:timers/promises')
+
+const Fastify = require('fastify')
+const { buildFederationSchema } = require('@mercuriusjs/federation')
+const GQL = require('mercurius')
+const plugin = require('../index')
+
+t.beforeEach(({ context }) => {
+  context.clock = FakeTimers.install({
+    shouldClearNativeTimers: true,
+    shouldAdvanceTime: true,
+    advanceTimeDelta: 40
+  })
+})
+
+t.afterEach(({ context }) => {
+  context.clock.uninstall()
+})
+
+test('Refreshing gateway schema', async (t) => {
```

Comment:
I assume this is already the case, but can you please make extra sure that this test fails without the change in the production code?

Reply:
Yes I wrote the test before making the fix to ensure it failed.